providers-sdk/v1/plugin/service.go

```diff
@@ -132,6 +141,7 @@ func (s *Service) Disconnect(req *DisconnectReq) (*DisconnectRes, error) {
 	s.runtimesLock.Lock()
 	defer s.runtimesLock.Unlock()
 	s.doDisconnect(req.Connection)
+	s.Flush() // flush our Memoizer
```

Comment:
something here doesn't feel quite right. You remove everything from the cache when any runtime disconnects. But there could still be other runtimes that are still running

Reply:
Good point. Should we check if there are still runtimes, and we only flush once we have no runtime?